src/components/functional_module/include/functional_module/timer/timer_director_impl.h

```diff
@@ -113,6 +113,7 @@ void TimerDirector::UnregisterTimer(const ModuleTimer<T>& timer) {
     return;
   }
   it->second->stop();
```

Comment:
Why didn't you use join?
If I remember correctly stop() only stops thread but it will be in system as dead thread. So we need to use join() to system can remove resources.